Framework/Source/Sample.cpp

```diff
@@ -40,504 +40,695 @@
+		std::string startTUVM_MB = std::to_string(memoryCheckRange.startCheck.pTotalUsedVirtualMemory / (1024 * 1024));
+		std::string startCUVM_MB = std::to_string(memoryCheckRange.startCheck.pCurrentlyUsedVirtualMemory / (1024 * 1024));
+
+		std::string startCheck = "At the Start Frame, " + std::to_string(memoryCheckRange.startCheck.pFrame) + ", : \n ";
+		startCheck = startCheck + ("Total Virtual Memory : " + startTVM_B + " bytes, " + startTVM_MB + " MB. \n");
+		startCheck = startCheck + ("Total Used Virtual Memory By All Processes : " + startTUVM_B + " bytes, " + startTUVM_MB + " MB. \n");
+		startCheck = startCheck + ("Virtual Memory used by this Process : " + startCUVM_B + " bytes, " + startCUVM_MB + " MB. \n \n");
+
+		//	Get the Strings for the Memory in Bytes - End Frame
+		std::string endTVM_B = std::to_string(memoryCheckRange.endCheck.pTotalVirtualMemory);
+		std::string endTUVM_B = std::to_string(memoryCheckRange.endCheck.pTotalUsedVirtualMemory);
+		std::string endCUVM_B = std::to_string(memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory);
+
+		std::string endTVM_MB = std::to_string(memoryCheckRange.endCheck.pTotalVirtualMemory / (1024 * 1024));
+		std::string endTUVM_MB = std::to_string(memoryCheckRange.endCheck.pTotalUsedVirtualMemory / (1024 * 1024));
+		std::string endCUVM_MB = std::to_string(memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory / (1024 * 1024));
+
+		std::string endCheck = "At the End Frame, " + std::to_string(memoryCheckRange.endCheck.pFrame) + ", : \n ";
+		endCheck = endCheck + ("Total Virtual Memory : " + endTVM_B + " bytes, " + endTVM_MB + " MB. \n");
+		endCheck = endCheck + ("Total Used Virtual Memory By All Processes : " + endTUVM_B + " bytes, " + endTUVM_MB + " MB. \n");
+		endCheck = endCheck + ("Virtual Memory used by this Process : " + endCUVM_B + " bytes, " + endCUVM_MB + " MB. \n \n");
+
+		//	Compute the Difference Between the Two.
+		std::string differenceCheck = "Difference : \n";
+		long long difference = 0;
+
+		{
+			difference = (long long)memoryCheckRange.endCheck.pCurrentlyUsedVirtualMemory - (long long)memoryCheckRange.startCheck.pCurrentlyUsedVirtualMemory;
+			differenceCheck = differenceCheck + std::to_string(difference) + "\n \n";
+		}
+
+		//	Key string for difference.
+		std::string keystring = std::to_string(memoryCheckRange.startCheck.pFrame) + " " + std::to_string(memoryCheckRange.endCheck.pFrame) + " " + (startCUVM_B)+" " + (endCUVM_B)+" " + std::to_string(difference) + " \n";
+
+		//	Get the name of the current program.
+		std::string filename = getExecutableName();
+
+		// Now we have a folder and a filename, look for an available filename (we don't overwrite existing files)
+		std::string prefix = std::string(filename) + ".MemoryFrameCheck";
+		std::string executableDir = getExecutableDirectory();
+		std::string txtFile;
+
+
+		//	Get an available filename.
+		if (findAvailableFilename(prefix, executableDir, "txt", txtFile))
```

Comment:
This is confusing. We use the same naming convention for log file. Give it a unique name (like prefix with 'mem' or something similar)